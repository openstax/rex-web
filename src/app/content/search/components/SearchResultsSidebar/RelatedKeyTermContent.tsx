import React from 'react';
import { KeyTermHit } from '../../types';
import * as Styled from './styled';

// tslint:disable-next-line: variable-name
const RelatedKeyTermContent = ({ keyTermHit }: { keyTermHit: KeyTermHit }) => <Styled.KeyTermResult>
  <Styled.KeyTerm>{keyTermHit.highlight.term}</Styled.KeyTerm>
  <div dangerouslySetInnerHTML={{ __html: keyTermHit.highlight.visibleContent[0] }} />
</Styled.KeyTermResult>;

export default RelatedKeyTermContent;
