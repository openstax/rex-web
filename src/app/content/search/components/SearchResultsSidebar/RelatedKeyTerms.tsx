import { SearchResultHit } from '@openstax/open-search-client';
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useDispatch } from 'react-redux';
import { assertDefined } from '../../../../utils/assertions';
import { Book } from '../../../types';
import { findArchiveTreeNodeById } from '../../../utils/archiveTreeUtils';
import { closeSearchResultsMobile } from '../../actions';
import { SelectedResult } from '../../types';
import SearchResultHits from './SearchResultHits';
import * as Styled from './styled';

// tslint:disable-next-line: variable-name
const RelatedKeyTerms = ({ book, keyTermHits, selectedResult }: {
  book: Book, keyTermHits: SearchResultHit[], selectedResult: SelectedResult | null
}) => {
  const dispatch = useDispatch();

  return <Styled.RelatedKeyTerms>
    <Styled.SearchResultsSectionTitle>
      <FormattedMessage id='i18n:search-results:related-key-term:title'>
        {(msg) => msg}
      </FormattedMessage>
    </Styled.SearchResultsSectionTitle>
    <SearchResultHits
      book={book}
      hits={keyTermHits}
      testId='related-key-term-result'
      getPage={(hit) => assertDefined(findArchiveTreeNodeById(book.tree, hit.source.pageId), 'hit has to be in a book')}
      onClick={() => dispatch(closeSearchResultsMobile())}
      selectedResult={selectedResult}
    />
  </Styled.RelatedKeyTerms>;
};

export default RelatedKeyTerms;
