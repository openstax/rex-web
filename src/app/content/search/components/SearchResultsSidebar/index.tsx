import React, { Component } from 'react';
import { connect } from 'react-redux';
import { AppState, Dispatch } from '../../../../types';
import * as select from '../../../selectors';
import { Book } from '../../../types';
import { clearSearch } from '../../actions';
import * as selectSearch from '../../selectors';
import { SearchResultContainer, SelectedResult } from '../../types';
import { SearchResultsBarWrapper } from './SearchResultsBarWrapper';

interface Props {
  book?: Book;
  query: string | null;
  hasQuery: boolean;
  totalHits: number | null;
  results: SearchResultContainer[] | null;
  onClose: () => void;
  searchResultsOpen: boolean;
  selectedResult: SelectedResult | null;
}

interface State {
  results: SearchResultContainer[] | null;
  query: string | null;
  totalHits: number | null;
  selectedResult: SelectedResult | null;
}

export class SearchResultsSidebar extends Component<Props, State> {

  public static getDerivedStateFromProps(newProps: Props, state: State) {
    if (newProps.results || (newProps.query !== state.query && newProps.query)) {
      return SearchResultsSidebar.getStateProps(newProps);
    }
    return state;
  }

  private static getStateProps(props: Props) {
    return {
      query: props.query,
      results: props.results,
      selectedResult: props.selectedResult,
      totalHits: props.totalHits,
    };
  }
  public state: State = {
    query: null,
    results: null,
    selectedResult: null,
    totalHits: null,
  };

  public constructor(props: Props) {
    super(props);

    this.state = SearchResultsSidebar.getStateProps(props);
  }

  public render() {
    return this.state.query ? <SearchResultsBarWrapper
      {...this.props}
      {...this.state}
      data-analytics-region='search-results'
    /> : null;
  }
}

export default connect(
  (state: AppState) => ({
    book: select.book(state),
    hasQuery: !!selectSearch.query(state),
    query: selectSearch.query(state),
    results: selectSearch.results(state),
    searchResultsOpen: selectSearch.searchResultsOpen(state),
    selectedResult: selectSearch.selectedResult(state),
    totalHits: selectSearch.totalHits(state),
  }),
  (dispatch: Dispatch) => ({
    onClose: () => {
      dispatch(clearSearch());
    },
  })
)(SearchResultsSidebar);
