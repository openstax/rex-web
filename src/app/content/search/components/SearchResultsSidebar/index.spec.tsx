import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import ReactTestUtils from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import SearchResultsSidebar from '.';
import createTestStore from '../../../../../test/createTestStore';
import {
  book as archiveBook,
  page,
  pageInChapter,
  pageInOtherChapter
} from '../../../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../../../test/mocks/osWebLoader';
import {
  makeEvent,
  makeFindByTestId,
  makeFindOrNullByTestId,
  renderToDom
} from '../../../../../test/reactutils';
import {
  makeSearchResultHit,
  makeSearchResults
} from '../../../../../test/searchResults';
import TestContainer from '../../../../../test/TestContainer';
import { Store } from '../../../../types';
import { assertDocument, assertWindow } from '../../../../utils';
import { receiveBook, receivePage } from '../../../actions';
import { formatBookData } from '../../../utils';
import * as domUtils from '../../../utils/domUtils';
import {
  clearSearch,
  closeSearchResultsMobile,
  receiveSearchResults,
  requestSearch,
  selectSearchResult
} from '../../actions';
import { SearchResultsBarWrapper } from './SearchResultsBarWrapper';

describe('SearchResultsSidebar', () => {
  let store: Store;
  let dispatch: jest.SpyInstance;

  const animationEvent = () => {
    return new (assertWindow().Event)('webkitAnimationEnd');
  };

  beforeEach(() => {
    store = createTestStore();
    dispatch = jest.spyOn(store, 'dispatch');
    store.dispatch(receiveBook(formatBookData(archiveBook, mockCmsBook)));
  });

  const render = () => (
    <TestContainer store={store}>
      <SearchResultsSidebar/>
    </TestContainer>
  );

  it('mounts and unmounts without a dom', () => {
    store.dispatch(requestSearch('cool search'));
    const component = renderer.create(render());
    expect(() => component.unmount()).not.toThrow();
  });

  it('mounts and unmmounts with a dom', () => {
    store.dispatch(requestSearch('cool search'));
    const { root } = renderToDom(render());
    expect(() => unmountComponentAtNode(root)).not.toThrow();
  });

  it('is initially null when there is no search', () => {
    const component = renderer.create(render());
    const findById = makeFindOrNullByTestId(component.root);

    expect(findById('search-results-sidebar')).toBe(null);
  });

  it('is hidden after search is cleared', () => {
    const component = renderer.create(render());

    renderer.act(() => {
      store.dispatch(requestSearch('cool search'));
      store.dispatch(receiveSearchResults(makeSearchResults()));
    });
    renderer.act(() => {
      store.dispatch(clearSearch());
    });

    const findById = makeFindByTestId(component.root);

    expect(findById('search-results-sidebar').props.searchResultsOpen).toBe(false);
  });

  it('shows sidebar with loading state if there is a search', () => {
    store.dispatch(requestSearch('cool search'));
    const component = renderer.create(render());
    const findById = makeFindByTestId(component.root);

    expect(() => findById('loader')).not.toThrow();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot for no search results', () => {
    store.dispatch(requestSearch('cool search'));
    store.dispatch(receiveSearchResults(makeSearchResults([])));

    const tree = renderer.create(render()).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with results', () => {
    store.dispatch(receivePage({ ...pageInChapter, references: [] }));
    store.dispatch(requestSearch('cool search'));
    const selectedResult = makeSearchResultHit({ book: archiveBook, page });
    store.dispatch(
      receiveSearchResults(
        makeSearchResults([
          selectedResult,
          makeSearchResultHit({ book: archiveBook, page: pageInChapter }),
          makeSearchResultHit({ book: archiveBook, page: pageInOtherChapter }),
        ])
      )
    );
    store.dispatch(selectSearchResult({result: selectedResult, highlight: 0}));

    const tree = renderer.create(render()).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('doesn\'t move focus when loading without results', () => {
    const activeElement = assertDocument().activeElement;
    renderToDom(render());

    expect(assertDocument().activeElement).toBe(activeElement);

    store.dispatch(receivePage({ ...page, references: [] }));
    store.dispatch(requestSearch('cool search'));
    store.dispatch(receiveSearchResults(makeSearchResults([])));

    expect(assertDocument().activeElement).toBe(activeElement);
  });

  it('closes search results when one is clicked', () => {
    store.dispatch(requestSearch('cool search'));
    store.dispatch(
      receiveSearchResults(
        makeSearchResults([makeSearchResultHit({ book: archiveBook, page })])
      )
    );

    const component = renderer.create(render());
    const findById = makeFindByTestId(component.root);

    expect(dispatch).not.toHaveBeenCalledWith(closeSearchResultsMobile());

    renderer.act(() => {
      findById('search-result').props.onClick(makeEvent());
    });

    expect(dispatch).toHaveBeenCalledWith(closeSearchResultsMobile());
  });

  it('clears search when closed', () => {
    store.dispatch(requestSearch('cool search'));
    store.dispatch(
      receiveSearchResults(
        makeSearchResults([makeSearchResultHit({ book: archiveBook, page })])
      )
    );

    const component = renderer.create(render());
    const findById = makeFindByTestId(component.root);

    expect(dispatch).not.toHaveBeenCalledWith(clearSearch());

    renderer.act(() => {
      findById('close-search').props.onClick(makeEvent());
    });

    expect(dispatch).toHaveBeenCalledWith(clearSearch());
  });

  it('fixes overscroll in safari', () => {
    const {tree} = renderToDom(render());
    const fixForSafariMock = jest.spyOn(domUtils, 'fixSafariScrolling');

    store.dispatch(requestSearch('cool search'));
    store.dispatch(receiveSearchResults(makeSearchResults()));

    const sidebar = ReactTestUtils.findRenderedComponentWithType(tree, SearchResultsBarWrapper);

    jest.useFakeTimers();
    if (sidebar.searchSidebar.current) {
      sidebar.searchSidebar.current.dispatchEvent(animationEvent());
    }
    jest.runAllTimers();

    expect(fixForSafariMock).toHaveBeenCalled();
  });
});
