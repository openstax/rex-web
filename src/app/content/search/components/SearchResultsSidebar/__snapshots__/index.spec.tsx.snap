// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsSidebar matches snapshot for no search results 1`] = `
<div
  aria-label="Search results sidebar"
  className="styled__SearchResultsBar-as3fh1-3 llsXIP"
  data-testid="search-results-sidebar"
>
  <div>
    <div
      className="styled__CloseIconWrapper-as3fh1-17 eZjRtu"
    >
      <button
        className="styled__CloseIconButton-as3fh1-16 ehlFTE"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="styled__CloseIcon-as3fh1-1 gsCQGC"
          focusable="false"
          height="15px"
          version="1.1"
          viewBox="0 0 15 15"
          width="15px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              transform="translate(-302.000000, -18.000000)"
            >
              <g
                transform="translate(302.000000, 18.000000)"
              >
                <path
                  d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
    </div>
    <div
      className="styled__SearchQuery-as3fh1-4 hPLPan"
    >
      <div
        className="styled__SearchQueryAlignment-as3fh1-15 zBfIM"
      >
        Sorry, no results found for 
         
        <strong>
           ‘
          cool search
          ’
        </strong>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchResultsSidebar matches snapshot with results 1`] = `
<div
  aria-label="Search results sidebar"
  className="styled__SearchResultsBar-as3fh1-3 llsXIP"
  data-testid="search-results-sidebar"
>
  <div
    className="styled__SearchQueryWrapper-as3fh1-5 gtPURI"
  >
    <div
      className="styled__SearchQuery-as3fh1-4 hPLPan"
    >
      <svg
        aria-hidden="true"
        className="styled__SearchIconInsideBar-as3fh1-0 dgtBTP StyledIconBase-sc-bdy9j4 cCvWXa"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 512 512"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
      <div
        className="styled__HeaderQuery-as3fh1-19 crkhDN"
      >
        3
         
        search results for 
        <strong>
           ‘
          cool search
          ’
        </strong>
      </div>
      <button
        className="styled__CloseIconButton-as3fh1-16 ehlFTE"
        data-testid="close-search"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="styled__CloseIcon-as3fh1-1 gsCQGC"
          focusable="false"
          height="15px"
          version="1.1"
          viewBox="0 0 15 15"
          width="15px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              transform="translate(-302.000000, -18.000000)"
            >
              <g
                transform="translate(302.000000, 18.000000)"
              >
                <path
                  d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
    </div>
  </div>
  <ol
    className="styled__NavOl-as3fh1-2 jPrtty"
  >
    <li
      className="styled__NavItem-as3fh1-14 kLrMDD"
    >
      <div
        className="styled__LinkWrapper-as3fh1-12 edDtdw"
      >
        <div
          className="styled__SearchResultsLink-as3fh1-10 hnAJVM"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<span class=\\"os-text\\">Test Page 1</span>",
            }
          }
        />
      </div>
      <a
        className="styled__SectionContentPreview-as3fh1-11 bJhIrl"
        data-testid="search-result"
        href="books/book-slug-1/pages/test-page-1?query=cool%20search&target=%7B%22index%22%3A0%2C%22type%22%3A%22search%22%7Dfs-id1544727"
        onClick={[Function]}
        result={
          Object {
            "highlight": 0,
            "result": Object {
              "highlight": Object {
                "visibleContent": Array [
                  "cool <strong>highlight</strong> bruh",
                ],
              },
              "index": "testbook1-uuid@1.0_i1",
              "score": 2,
              "source": Object {
                "elementId": "fs-id1544727",
                "elementType": "paragraph",
                "pageId": "testbook1-testpage1-uuid@1.0",
                "pagePosition": 60,
              },
            },
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "cool <strong>highlight</strong> bruh",
            }
          }
          tabIndex={-1}
        />
      </a>
    </li>
    <li
      className="styled__ListItem-as3fh1-20 bgLAKc"
    >
      <details
        className="Details-xi326o-3 styled__Details-as3fh1-7 jLcYfE"
        open={true}
      >
        <summary
          className="Details__Summary-xi326o-2 styled__SearchBarSummary-as3fh1-9 hPtQFE"
          tabIndex={0}
        >
          <div
            className="styled__SearchBarSummaryContainer-as3fh1-8 eSHSds"
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              className="Details__ExpandIcon-xi326o-0 cVuUUi StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 192 512"
            >
              <path
                d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="Details__CollapseIcon-xi326o-1 jsdrhW StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 320 512"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                fill="currentColor"
              />
            </svg>
            <span
              className="styled__SummaryTitle-as3fh1-6 kQeKbP"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"os-number\\">10</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 5</span>",
                }
              }
            />
          </div>
        </summary>
        <ol
          className="styled__DetailsOl-as3fh1-13 ggpsqK"
        >
          <li
            className="styled__NavItem-as3fh1-14 kLrMDD"
          >
            <div
              aria-label="Current Page"
              className="styled__LinkWrapper-as3fh1-12 edDtdw"
            >
              <div
                className="styled__SearchResultsLink-as3fh1-10 hnAJVM"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"os-text\\">Test Page 6</span>",
                  }
                }
              />
            </div>
            <a
              className="styled__SectionContentPreview-as3fh1-11 fhVukS"
              data-testid="search-result"
              href="books/book-slug-1/pages/10-test-page-6?query=cool%20search&target=%7B%22index%22%3A0%2C%22type%22%3A%22search%22%7Dfs-id1544727"
              onClick={[Function]}
              result={
                Object {
                  "highlight": 0,
                  "result": Object {
                    "highlight": Object {
                      "visibleContent": Array [
                        "cool <strong>highlight</strong> bruh",
                      ],
                    },
                    "index": "testbook1-uuid@1.0_i1",
                    "score": 2,
                    "source": Object {
                      "elementId": "fs-id1544727",
                      "elementType": "paragraph",
                      "pageId": "testbook1-testpage6-uuid@1.0",
                      "pagePosition": 60,
                    },
                  },
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "cool <strong>highlight</strong> bruh",
                  }
                }
                tabIndex={-1}
              />
            </a>
          </li>
        </ol>
      </details>
    </li>
    <li
      className="styled__ListItem-as3fh1-20 bgLAKc"
    >
      <details
        className="Details-xi326o-3 styled__Details-as3fh1-7 jLcYfE"
        open={true}
      >
        <summary
          className="Details__Summary-xi326o-2 styled__SearchBarSummary-as3fh1-9 hPtQFE"
          tabIndex={0}
        >
          <div
            className="styled__SearchBarSummaryContainer-as3fh1-8 eSHSds"
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              className="Details__ExpandIcon-xi326o-0 cVuUUi StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 192 512"
            >
              <path
                d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              className="Details__CollapseIcon-xi326o-1 jsdrhW StyledIconBase-sc-bdy9j4 cCvWXa"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 320 512"
            >
              <path
                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                fill="currentColor"
              />
            </svg>
            <span
              className="styled__SummaryTitle-as3fh1-6 kQeKbP"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<span class=\\"os-number\\">12</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 6</span>",
                }
              }
            />
          </div>
        </summary>
        <ol
          className="styled__DetailsOl-as3fh1-13 ggpsqK"
        >
          <li
            className="styled__NavItem-as3fh1-14 kLrMDD"
          >
            <div
              className="styled__LinkWrapper-as3fh1-12 edDtdw"
            >
              <div
                className="styled__SearchResultsLink-as3fh1-10 hnAJVM"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"os-text\\">Test Page 7</span>",
                  }
                }
              />
            </div>
            <a
              className="styled__SectionContentPreview-as3fh1-11 fhVukS"
              data-testid="search-result"
              href="books/book-slug-1/pages/12-test-page-7?query=cool%20search&target=%7B%22index%22%3A0%2C%22type%22%3A%22search%22%7Dfs-id1544727"
              onClick={[Function]}
              result={
                Object {
                  "highlight": 0,
                  "result": Object {
                    "highlight": Object {
                      "visibleContent": Array [
                        "cool <strong>highlight</strong> bruh",
                      ],
                    },
                    "index": "testbook1-uuid@1.0_i1",
                    "score": 2,
                    "source": Object {
                      "elementId": "fs-id1544727",
                      "elementType": "paragraph",
                      "pageId": "testbook1-testpage7-uuid@1.0",
                      "pagePosition": 60,
                    },
                  },
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "cool <strong>highlight</strong> bruh",
                  }
                }
                tabIndex={-1}
              />
            </a>
          </li>
        </ol>
      </details>
    </li>
  </ol>
</div>
`;

exports[`SearchResultsSidebar shows sidebar with loading state if there is a search 1`] = `
<div
  aria-label="Search results sidebar"
  className="styled__SearchResultsBar-as3fh1-3 llsXIP"
  data-testid="search-results-sidebar"
>
  <div
    aria-label="Loading results"
    className="styled__LoadingWrapper-as3fh1-18 bCoJZF"
  >
    <div
      className="styled__CloseIconWrapper-as3fh1-17 eZjRtu"
    >
      <button
        className="styled__CloseIconButton-as3fh1-16 ehlFTE"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="styled__CloseIcon-as3fh1-1 gsCQGC"
          focusable="false"
          height="15px"
          version="1.1"
          viewBox="0 0 15 15"
          width="15px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="currentColor"
              transform="translate(-302.000000, -18.000000)"
            >
              <g
                transform="translate(302.000000, 18.000000)"
              >
                <path
                  d="M7.5,5.41522791 L12.0331524,0.579865364 C12.3077536,0.286957429 12.7165503,0.24816296 12.946282,0.493210121 L13.9861449,1.60239723 C14.2158766,1.84744439 14.1795068,2.28349422 13.9049056,2.57640216 L9.37175324,7.41176471 L13.9049056,12.2471273 C14.1795068,12.5400352 14.2158766,12.976085 13.9861449,13.2211322 L12.946282,14.3303193 C12.7165503,14.5753665 12.3077536,14.536572 12.0331524,14.243664 L7.5,9.4083015 L2.96684761,14.243664 C2.69224642,14.536572 2.2834497,14.5753665 2.05371799,14.3303193 L1.01385508,13.2211322 C0.784123363,12.976085 0.820493178,12.5400352 1.09509437,12.2471273 L5.62824676,7.41176471 L1.09509437,2.57640216 C0.820493178,2.28349422 0.784123363,1.84744439 1.01385508,1.60239723 L2.05371799,0.493210121 C2.2834497,0.24816296 2.69224642,0.286957429 2.96684761,0.579865364 L7.5,5.41522791 Z"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
    </div>
    <svg
      className="Loader__LoadingIcon-dfuqve-0 ceHiLy"
      data-testid="loader"
      id="Layer_1"
      version="1.1"
      viewBox="0 0 57.6 39.1"
      x="0px"
      y="0px"
    >
      <path
        className="os-blue"
        d="M38.4,22c0,0.4-0.3,0.7-0.7,0.7H13.5c-0.4,0-0.7-0.3-0.7-0.7v-1.4c0-0.4,0.3-0.7,0.7-0.7h24.2 c0.4,0,0.7,0.3,0.7,0.7V22z"
        id="os-blue"
        style={
          Object {
            "fill": "#00246a",
          }
        }
      />
      <path
        className="os-yellow"
        d="M41.8,17.8c0,0.4-0.3,0.7-0.7,0.7l-25.4,0.8c-0.4,0-0.7-0.3-0.7-0.7l-0.1-2.1 c0-0.4,0.3-0.7,0.7-0.7L41,15c0.4,0,0.7,0.3,0.7,0.7L41.8,17.8z"
        id="os-yellow"
        style={
          Object {
            "fill": "#f4d019",
          }
        }
      />
      <path
        className="os-gray"
        d="M34.8,14c0,0.4-0.3,0.7-0.7,0.7H11.6c-0.4,0-0.7-0.3-0.7-0.7v-3.5c0-0.4,0.3-0.7,0.7-0.7h22.5 c0.4,0,0.7,0.3,0.7,0.7V14z"
        id="os-gray"
        style={
          Object {
            "fill": "#5e6062",
          }
        }
      />
      <path
        className="os-orange"
        d="M39,8.1c0,0.3-0.4,0.5-1,0.5H8.6c-0.5,0-1-0.2-1-0.5V6.9c0-0.3,0.4-0.5,1-0.5H38 c0.5,0,1,0.2,1,0.5V8.1z"
        id="os-orange"
        style={
          Object {
            "fill": "#f47641",
          }
        }
      />
      <path
        className="os-green"
        d="M43.9,5c0,0.7-0.6,1.2-1.2,1.1L15,4.8c-0.7,0-1.2-0.6-1.1-1.2L14,1.1c0-0.7,0.6-1.2,1.2-1.1 l27.7,1.3c0.7,0,1.2,0.6,1.1,1.2L43.9,5z"
        id="os-green"
        style={
          Object {
            "fill": "#77af42",
          }
        }
      />
      <path
        className="os-type"
        d="M3.1,30.2c-1.8,0-3.1,1.3-3.1,3.1c0,1.8,1.3,3.1,3.1,3.1c1.8,0,3.1-1.3,3.1-3.1 C6.2,31.5,4.9,30.2,3.1,30.2z M3.1,35.7c-1.4,0-2.3-1.1-2.3-2.4c0-1.3,0.9-2.4,2.3-2.4c1.4,0,2.3,1.1,2.3,2.4 C5.4,34.6,4.5,35.7,3.1,35.7z M10.9,30.2c-0.9,0-1.9,0.4-2.4,1.2h0v-1H7.7v8.8h0.8v-3.9h0C9,36,10,36.4,10.9,36.4 c1.8,0,3.1-1.3,3.1-3.1C14,31.5,12.7,30.2,10.9,30.2z M10.9,35.7c-1.4,0-2.5-1.1-2.5-2.4c0-1.3,1.1-2.4,2.5-2.4 c1.4,0,2.3,1.1,2.3,2.4C13.2,34.6,12.3,35.7,10.9,35.7z M18.3,30.2c-1.7,0-2.9,1.3-2.9,3.1c0,1.8,1.2,3.1,3,3.1c1,0,2-0.4,2.6-1.3 l-0.6-0.5c-0.4,0.6-1.2,1-2,1c-1.8,0-2.2-1.6-2.2-2.2h5v-0.5C21.1,31.6,20.1,30.2,18.3,30.2z M16.1,32.8c0-0.3,0.5-1.9,2.1-1.9 c1.1,0,2,0.9,2,1.9H16.1z M27.9,32.7v3.5h-0.8v-3.5c0-1-0.3-1.9-1.5-1.9c-1.1,0-2,0.8-2,2.4v2.9h-0.8V32c0-0.4-0.1-1.4-0.1-1.7h0.8 c0,0.4,0,0.9,0.1,1h0c0.3-0.7,1.1-1.2,2-1.2C27.4,30.2,27.9,31.4,27.9,32.7z M34.2,34.4c0,1.5-1.4,2-2.6,2c-0.9,0-1.8-0.2-2.4-0.9 l1-1c0.4,0.4,0.8,0.8,1.5,0.8c0.4,0,1-0.2,1-0.7c0-1.3-3.3-0.3-3.3-2.6c0-1.4,1.2-2,2.5-2c0.8,0,1.7,0.3,2.2,0.9l-1,0.9 c-0.3-0.4-0.7-0.6-1.2-0.6c-0.4,0-0.9,0.2-0.9,0.7C30.9,32.9,34.2,32,34.2,34.4z M37.5,30.1h1.7v1.3h-1.7v2.7c0,0.6,0.2,1,0.9,1 c0.3,0,0.6-0.1,0.8-0.2v1.3c-0.3,0.2-0.9,0.2-1.2,0.2c-1.6,0-2-0.7-2-2.2v-2.8h-1.3v-1.3H36v-1.8h1.5V30.1z M43,30 c-1,0-1.9,0.3-2.6,1l0.8,0.8c0.4-0.4,1-0.6,1.6-0.6c0.8,0,1.4,0.4,1.4,1.1v0.2h-0.4c-1.5,0-3.9,0.1-3.9,2.1c0,1.2,1.1,1.8,2.2,1.8 c0.8,0,1.5-0.3,2-1h0v0.8h1.4v-3.5C45.6,31.8,45.6,30,43,30z M44.1,33.8c0,0.9-0.5,1.4-1.5,1.4c-0.5,0-1.1-0.2-1.1-0.8 c0-0.9,1.5-0.9,2.3-0.9h0.3V33.8z M50.8,32.9l2.5,3.3h-1.9L49.8,34l-1.6,2.2h-1.8l2.5-3.3l-2.1-2.8h1.9l1.2,1.8l1.3-1.8h1.7 L50.8,32.9z M57.6,30.1v1.7h-0.3v-1.5h0l-0.6,1.5h-0.2l-0.6-1.5h0v1.5h-0.3v-1.7h0.4l0.5,1.3l0.5-1.3H57.6z M53.9,30.1h1.4v0.3h-0.5 v1.5h-0.3v-1.5h-0.5V30.1z"
        id="os-type"
        style={
          Object {
            "fill": "#5e6062",
          }
        }
      />
    </svg>
  </div>
</div>
`;
